From 08f7ea36809aa3d156a118e6a6a0c2f03f4ce3d8 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Fri, 10 Apr 2026 14:12:37 +0200
Subject: [PATCH] Only link with MATH/ZLib libraries if they're found

If they're not found they may not be required, or linking will fail.
---
 src/CMakeLists.txt | 22 ++++++++++++++--------
 1 file changed, 14 insertions(+), 8 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 09b5160..d426931 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -6,12 +6,9 @@ configure_file(config.h.in config.h)
 if(NOT MSVC)
   if(NOT WIN32)
     find_library(MATH m)
-  else()
-    set(MATH "")
   endif()
   include(FindZLIB)
 else()
-  set(MATH "")
   find_program(NUGET nuget)
   if(NUGET)
     execute_process(COMMAND ${NUGET} install zlib)
@@ -23,15 +20,12 @@ endif()
 if(NOT MSVC)
   if(NOT WIN32)
     find_library(MATH m)
-  else()
-    set(MATH "")
   endif()
   include(FindZLIB)
   include_directories(${ZLIB_INCLUDE_DIRS})
   set(PKG_CONFIG_PRIVATELIBS "-lm ${PKG_CONFIG_PRIVATELIBS}")
   set(PKG_CONFIG_PRIVATELIBS "-lz ${PKG_CONFIG_PRIVATELIBS}")
 else()
-  set(MATH "")
   find_program(NUGET nuget)
   if(NOT NUGET)
     message(
@@ -78,7 +72,12 @@ target_include_directories(mysofa-static
   "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/hrtf>"
   $<INSTALL_INTERFACE:include/>
 )
-target_link_libraries(mysofa-static LINK_PRIVATE ${MATH} ${ZLIB_LIBRARIES})
+if(MATH)
+    target_link_libraries(mysofa-static LINK_PRIVATE ${MATH})
+endif()
+if(ZLIB_LIBRARIES)
+    target_link_libraries(mysofa-static LINK_PRIVATE ${ZLIB_LIBRARIES})
+endif()
 set_target_properties(
   mysofa-static
   PROPERTIES OUTPUT_NAME mysofa CLEAN_DIRECT_OUTPUT 1 POSITION_INDEPENDENT_CODE
@@ -110,7 +109,12 @@ if(BUILD_SHARED_LIBS)
     "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/hrtf>"
     $<INSTALL_INTERFACE:include/>
   )
-  target_link_libraries(mysofa-shared PRIVATE ${MATH} ${ZLIB_LIBRARIES})
+  if(MATH)
+    target_link_libraries(mysofa-shared PRIVATE ${MATH})
+  endif()
+  if(ZLIB_LIBRARIES)
+    target_link_libraries(mysofa-shared PRIVATE ${ZLIB_LIBRARIES})
+  endif()
   set_target_properties(mysofa-shared
                         PROPERTIES OUTPUT_NAME mysofa CLEAN_DIRECT_OUTPUT 1)
   # Rename the Windows import library (stub) of the DLL to prevent a name clash
